client/components/seo/meta-title-editor/index.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component, PropTypes } from 'react';
+import get from 'lodash/get';
+import identity from 'lodash/identity';
+import isString from 'lodash/isString';
+import isUndefined from 'lodash/isUndefined';
+import map from 'lodash/map';
+import { connect } from 'react-redux';
+
+import SegmentedControl from 'components/segmented-control';
+import TokenField from 'components/token-field';
+import { localize } from 'i18n-calypso';
+
+import {
+	nativeToRaw,
+	rawToNative,
+	removeBlanks
+} from './mappings';
+
+const titleTypes = translate => [
+	{ value: 'frontPage', label: translate( 'Front Page' ) },
+	{ value: 'posts', label: translate( 'Posts' ) },
+	{ value: 'pages', label: translate( 'Pages' ) },
+	{ value: 'groups', label: translate( 'Categories & Tags' ) },
+	{ value: 'archives', label: translate( 'Archives' ) }
+];
+
+const getValidTokens = translate => ( {
+	siteName: translate( 'Site Name' ),
+	tagline: translate( 'Tagline' ),
+	postTitle: translate( 'Post Title' ),
+	pageTitle: translate( 'Page Title' ),
+	groupTitle: translate( 'Category/Tag Title' ),
+	date: translate( 'Date' )
+} );
+
+const tokenMap = {
+	frontPage: [ 'siteName', 'tagline' ],
+	posts: [ 'siteName', 'tagline', 'postTitle' ],
+	pages: [ 'siteName', 'tagline', 'pageTitle' ],
+	groups: [ 'siteName', 'tagline', 'groupTitle' ],
+	archives: [ 'siteName', 'tagline', 'date' ]
+};
+
+const tokenize = translate => s => {
```

Comment:
Do you think that adding characters here will improve the legibility? I didn't add them because I didn't think it would. `s` would become `string`, `k` would become `key`, but those names are basically already surrounded with the context indicating such, and making them longer didn't look as clear to me.